src/Components/Medicine/AdministerMedicine.tsx

```diff
@@ -41,21 +45,47 @@ export default function AdministerMedicine({ prescription, ...props }: Props) {
       description={
         <div className="text-sm font-semibold leading-relaxed text-gray-600">
           <CareIcon className="care-l-history-alt pr-1" /> Last administered
-          <span className="pl-1">
-            {prescription.last_administered_on
-              ? formatDateTime(prescription.last_administered_on)
+          <span className="whitespace-nowrap pl-2">
+            <CareIcon className="care-l-clock" />{" "}
+            {prescription.last_administration?.administered_date
+              ? formatDateTime(
+                  prescription.last_administration.administered_date
+                )
               : t("never")}
           </span>
+          {prescription.dosage_type === "TITRATED" && (
+            <span className="whitespace-nowrap pl-2">
+              <CareIcon className="care-l-syringe" /> {t("dosage")}
+              {":"} {prescription.last_administration?.dosage ?? "NA"}
+            </span>
+          )}
+          <span className="whitespace-nowrap pl-2">
+            <CareIcon className="care-l-user" /> Administered by:{" "}
+            {prescription.last_administration?.administered_by?.username ??
+              "NA"}
+          </span>
         </div>
       }
       show
       onClose={() => props.onClose(false)}
       onConfirm={async () => {
+        if (prescription.dosage_type === "TITRATED") {
+          setError(
+            AdministrationDosageValidator(
+              dosage,
+              prescription.base_dosage,
+              prescription.target_dosage
+            )
+          );
+          return;
```

Comment:
Since you are always returning, titrated prescriptions would never be administered right?

Reply:
I didn't even notice this my bad.
Great find btw!

Reply:
You should be testing your changes btw 😄